bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,609 @@
+                LOG.warn("Failed session.logout()", e);
+            }
+        }
+
+        if (this.resourceResolver != null) {
+            try {
+                this.resourceResolver.close();
+            } catch (Exception e) {
+                LOG.warn("Failed resourceResolver.close()", e);
+            }
+        }
+    }
+
+    /**
+     * @see RemoteAssetsNodeSync#syncAssetNodes()
+     */
+    @Override
+    public void syncAssetNodes() {
+        List<String> syncPaths = new ArrayList<>();
+        syncPaths.addAll(this.remoteAssetsConfig.getTagSyncPaths());
+        syncPaths.addAll(this.remoteAssetsConfig.getDamSyncPaths());
+
+        try {
+            for (String syncPath : syncPaths) {
+                this.session.refresh(true);
+                JSONObject topLevelJson = getJsonFromUri(syncPath);
+                Node topLevelSyncNode = getOrCreateNode(syncPath, (String) topLevelJson.get(JcrConstants.JCR_PRIMARYTYPE));
+                createOrUpdateNodes(topLevelJson, topLevelSyncNode);
+                this.session.save();
+            }
+        } catch (IOException e) {
+            LOG.error("IO Exception {}", e);
+        } catch (JSONException e) {
+            LOG.error("Json Exception {}", e);
+        } catch (RepositoryException e) {
+            LOG.error("Repository Exception {}", e);
+        }
+    }
+
+    /**
+     * Grab JSON from the provided path and sync to the JCR.
+     * @param nextPath String
+     * @param primaryType String
+     * @return Node
+     */
+    private Node getOrCreateNode(final String nextPath, final String primaryType) {
+        Node node = null;
+
+        try {
+            node = JcrUtils.getNodeIfExists(nextPath, this.session);
+            if (node == null) {
+                node = JcrUtil.createPath(nextPath, primaryType, this.session);
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("New node '{}' created.", node.getPath());
+                }
+            } else {
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("Node '{}' retrieved from JCR.", node.getPath());
+                }
+            }
+        } catch (RepositoryException e) {
+            LOG.error("Repository Exception. Unable to get or create node '{}'", nextPath, e);
+        }
+
+        return node;
+    }
+
+    /**
+     * Get {@link JSONObject} from URL response.
+     * @param path String
+     * @return JSONObject
+     * @throws IOException exception
+     * @throws JSONException exception
+     */
+    private JSONObject getJsonFromUri(final String path) throws IOException, JSONException {
+        try {
+            URI pathUi = new URI(null, null, path, null);
+            // we want to traverse the JCR one level at a time, hence the '1' selector.
+            URL url = new URL(this.remoteAssetsConfig.getServer().concat(pathUi.toString()).concat(".1.json"));
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
```

Comment:
As mentioned in other binary sync impl, we can probably pull a few of these lines out to a common function.